### DISCUSSION TITLE
How to source a bash file without sourcing bash startup files

### DISCUSSION THREAD

**Discussion by halloleo** [xonsh/xonsh#4990] [Q&A]
I have a bash file which sets a few environment variables. Sometimes I need to source in a xonsh session.

When I execute in xonsh:

    $ source-bash my-bash-file.sh

I get lots of alias error messages relating to stuff which is in my bash startup files (like `.bashrc`), not in the file I want to source (`my-bash-file.sh`). I guess `source-bash` by default sources first the startup files and then `my-bash-file.sh`, but I want to _only_ source `my-bash-file.sh`, nothing else.

How can I tell `source-bash` so? I looked at `source-bash --help` but didn't find a CLI option for this. Am I blind?

Thanks for pointers!

**Comment by gforsyth** [comment]
Hey @halloleo -- you aren't blind, it can be a bit tricky to work around.
There are two things required (since we _have_ to start a bash process in order to do the `source-bash`):

1. At the top of your `bashrc`, include this snippet:

```bash
[[ $- != *i* ]] && return
```

that exits your `rc` file if `bash` is not being run in interactive mode.

2. Call `source-bash` with the `-n` flag, to tell it to source in non-interactive mode.  Then `bash` will skip the `bashrc` file because of step 1, and only source the file you care about.
